src/Compilers/Core/Portable/DiagnosticAnalyzer/ShadowCopyAnalyzerAssemblyLoader.cs

```diff
@@ -53,6 +53,10 @@ public ShadowCopyAnalyzerAssemblyLoader(string baseDirectory = null)
 
         private void DeleteLeftoverDirectories()
         {
+            // Avoid first chance exception
+            if (!Directory.Exists(_baseDirectory))
+                return;
```

Comment:
Why the `catch` after a few lines won't simply achieve the same result?

Reply:
As mentioned in the description, this avoids the first chance exception (exception that is thrown but handled) which is painful while debugging.

Reply:
@Youssef1313 

The `catch` is still necessary because `Exists` does not actually tell u that a file does or does not exist. It tells u that a file, that you had access to, used to exist. By the time the `Exist` check returns that directory may be deleted, the USB drive it existed on was yanked from the machine, etc ... 

You can never predict what the file system will do, you can only react to it failing. 

https://blog.paranoidcoding.com/2009/12/10/the-file-system-is-unpredictable.html